modules/htlc/client/cli/tx.go

```diff
@@ -0,0 +1,213 @@
+package cli
+
+import (
+	"crypto/rand"
+	"encoding/hex"
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/context"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth"
+	"github.com/cosmos/cosmos-sdk/x/auth/client/utils"
+
+	"github.com/irisnet/irishub/modules/htlc/internal/types"
+)
+
+// GetTxCmd returns the transaction commands for this module
+func GetTxCmd(cdc *codec.Codec) *cobra.Command {
+	stakingTxCmd := &cobra.Command{
+		Use:                        types.ModuleName,
+		Short:                      "HTLC subcommands",
+		DisableFlagParsing:         true,
+		SuggestionsMinimumDistance: 2,
+		RunE:                       client.ValidateCmd,
+	}
+
+	stakingTxCmd.AddCommand(client.PostCommands(
+		GetCmdCreateHTLC(cdc),
+		GetCmdClaimHTLC(cdc),
+		GetCmdRefundHTLC(cdc),
+	)...)
+
+	return stakingTxCmd
+}
+
+// GetCmdCreateHTLC implements the create HTLC command
+func GetCmdCreateHTLC(cdc *codec.Codec) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create",
+		Short: "Create an HTLC",
+		Example: "iriscli tx htlc create --chain-id=<chain-id> --from=<key-name>" +
+			" --fees=0.3iris --to=<to> --receiver-on-other-chain=<receiver-on-other-chain> --amount=<amount>" +
+			" --secret=<secret> --timestamp=<timestamp> --time-lock=<time-lock>",
+		PreRunE: preCheckCmd,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			txBldr := auth.NewTxBuilderFromCLI().WithTxEncoder(auth.DefaultTxEncoder(cdc))
+			cliCtx := context.NewCLIContext().WithCodec(cdc)
+
+			sender := cliCtx.GetFromAddress()
+
+			toAddrStr := viper.GetString(FlagTo)
+			toAddr, err := sdk.AccAddressFromBech32(toAddrStr)
+			if err != nil {
+				return err
+			}
+
+			receiverOnOtherChain := viper.GetString(FlagReceiverOnOtherChain)
+
+			amountStr := viper.GetString(FlagAmount)
+			amount, err := sdk.ParseCoins(amountStr)
+			if err != nil {
+				return err
+			}
+
+			timestamp := viper.GetInt64(FlagTimestamp)
+			timeLock := viper.GetInt64(FlagTimeLock)
+
+			secret := make(types.HTLCSecret, 32)
+			var hashLock types.HTLCHashLock
+
+			flags := cmd.Flags()
+			if flags.Changed(FlagHashLock) {
+				hashLockStr := strings.TrimSpace(viper.GetString(FlagHashLock))
+				hashLock, err = hex.DecodeString(hashLockStr)
+				if err != nil {
+					return err
+				}
+			} else {
+				secretStr := strings.TrimSpace(viper.GetString(FlagSecret))
+				if len(secretStr) > 0 {
+					if len(secretStr) != 2*types.SecretLength {
+						return fmt.Errorf("the secret must be %d bytes long", types.SecretLength)
```

Comment:
"the secret must be %d bytes long" should be  "the length of the secret bytes must be %d"